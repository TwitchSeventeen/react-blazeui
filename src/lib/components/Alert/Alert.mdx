---
name: Alert
---

import { Playground, PropsTable } from 'docz'
import Alert from './index';

# Alert

## Properties
<PropsTable of={Alert} />

## Basic usage
<Playground>
    <Alert>lorem</Alert>
</Playground>

## Alert colours
Use the **type** prop to provide styling to the alert.
<Playground>
    <Alert>Default</Alert>
    <Alert type="brand">Brand</Alert>
    <Alert type="info">Brand</Alert>
    <Alert type="warning">Warning</Alert>
    <Alert type="success">Success</Alert>
    <Alert type="error">Error</Alert>
</Playground>

## Dismissable alerts
Use the **dismissable** prop to add a close button to the alert. The **onDismiss** prop is a function called when the user clicks it.
<Playground>
    <Alert type="info" dismissable onDismiss={this.onDismiss}>Brand</Alert>
</Playground>