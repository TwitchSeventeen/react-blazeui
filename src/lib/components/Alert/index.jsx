import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * An alert message
 * @param props
 * @returns {*}
 * @constructor
 */
function Alert (props) {
  const { children, dismissable, onDismiss, type } = props;
  const classes = classNames('c-alert', { [`c-alert--${type}`]: type });

  return (
    <div role="alert" className={classes}>
      {children}
      { dismissable &&
        <button className="c-button c-button--close" onClick={onDismiss} type="button">&times;</button>
      }
    </div>
  )
}

Alert.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * User can dismiss the alert
   */
  dismissable: PropTypes.bool,
  /**
   * Function called when user dismisses the alert
   */
  onDismiss: PropTypes.func,
  /**
   * The type of button
   */
  type: PropTypes.oneOf([
      'brand',
      'info',
      'warning',
      'success',
      'error',
  ])
};

Alert.defaultProps = {
  children: null,
  dismissable: false,
  onDismiss: null,
};

export default Alert;