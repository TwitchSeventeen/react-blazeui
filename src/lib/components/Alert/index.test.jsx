import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Alert from './index';

describe('Alert', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Alert).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Alert>lorem</Alert>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly a dismissable success alert', () => {
    const { asFragment } = render(
      <Alert
        dismissable
        onDismiss={jest.fn()}
        type="success"
      >
        lorem
      </Alert>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
