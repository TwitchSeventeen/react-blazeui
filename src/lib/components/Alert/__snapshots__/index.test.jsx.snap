// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<DocumentFragment>
  <div
    class="c-alert"
    data-name="alert"
    role="alert"
  >
    lorem
  </div>
</DocumentFragment>
`;

exports[`Alert renders correctly a dismissable success alert 1`] = `
<DocumentFragment>
  <div
    class="c-alert c-alert--success"
    data-name="alert"
    role="alert"
  >
    lorem
    <button
      class="c-button c-button--close"
      type="button"
    >
      Ã—
    </button>
  </div>
</DocumentFragment>
`;
